assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -399,7 +399,36 @@ public List<?> getAssignmentsForSite(EntityView view,
             throw new EntityNotFoundException("No access to site: " + siteId, siteId);
         }
 
-        assignmentService.getAssignmentsForContext(siteId).stream().map(SimpleAssignment::new).forEach(rv::add);
+        assignmentService.getAssignmentsForContext(siteId).stream()
+        .map(a->{
+            SimpleAssignment as = null;
+            try {
+                Assignment asgn = assignmentService.getAssignment(a.getId());
```

Comment:
this fetches the assignment a second time?

Reply:
Solved